main.tf

```diff
@@ -4,12 +4,14 @@ locals {
 
   db_rules = "${null_resource.subnet_mappings.*.triggers}"
 
-  vaultName = "infra-vault-${var.subscription}"
+  vaultName = var.key_vault_name != null ? var.key_vault_name : "infra-vault-${var.subscription}"
 }
 
 data "azurerm_key_vault" "infra_vault" {
-  name                = "${local.vaultName}"
-  resource_group_name = "${var.env == "prod" || var.env == "idam-prod" || var.env == "idam-prod2" ? "core-infra-prod" : "cnp-core-infra"}"
+  name = "${local.vaultName}"
+  resource_group_name = var.key_vault_rg != null ? var.key_vault_rg : (
```

Comment:
can you convert this to a local variable please so it sits next to the vault name